### PR TITLE
Introduce big endian types and EthHeader

### PR DESCRIPTION
**Pull request by sangjinhan**
This PR adds C++-style declaration of beXX_t type and Ethernet header, which used to rely on DPDK's rte_ether.h and rte_byteorder.h.

### CODE REVIEW

File: core/utils/endian.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#ifndef BESS_UTILS_ENDIAN_H
+#define BESS_UTILS_ENDIAN_H
+
+#include <cstdint>
+
+namespace bess {
+namespace utils {
+
+constexpr bool is_be_system() {
```
Comment (barath) [inline core/utils/endian.h]: Maybe expand "be" to "big_endian"?

File: core/utils/endian.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#ifndef BESS_UTILS_ENDIAN_H
```
Comment (melvinw) [inline core/utils/endian.h]: Nit: missing trailing _ here

File: core/utils/ether.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#ifndef BESS_UTILS_ETHER_H
```
Comment (melvinw) [inline core/utils/ether.h]: Need trailing _ here, too

File: core/utils/ether.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+struct [[gnu::packed]] EthHeader {
+  struct [[gnu::packed]] Address {
+    Address() = default;
+    Address(const std::string &str);
+
+    static const size_t kSize = 6;
+
+    // Parses str in "aA:Bb:00:11:22:33" format and saves the address in parsed
+    // Returns false if the format is incorrect.
+    //   (in that case, the content of parsed is undefined.)
```
Comment (barath) [inline core/utils/ether.h]: In this comment, what is "parsed" that is left undefined?  (I assume "this".)

**Comment by melvinw** [comment]
Looks good! Just minor nits

**Comment by sangjinhan** [comment]
I'll reflect Barath's comments to a next PR.

**Pull request merged**